### ISSUE TITLE
Cursor.execute_script

### ISSUE THREAD

**Issue by id-ilych** [PyMySQL/PyMySQL#326]
Please add method to execute multi-query script. It's not as trivial to implement it so I think it's worth to include that to library.

Difficulties: 
1. script file contains multiple commands which should be split in order to be passed to Cursor.execute
2. queries could be multiline
3. line could be commented out ('--')
4. part of the line could be commented out
5. comment could be somewhere in the middle of multiline query

Script example that shows them:
```
-- remove item
DELETE FROM Items WHERE typeid = 243; 
UPDATE Items
SET reg1i = (~(1 << 25)) & reg1i  -- reset 25th bit
WHERE typeid = 63 and state = 3600;
```

Thanks!

**Comment by methane** [comment]
Could you try this?
```python
def execute_script(conn, script):
    cur = conn.cursor()
    cur.execute(script)
    cur.close()
```

**Comment by id-ilych** [comment]
Hmm.. it does work in some cases. Here's problematic code:

```
import pymysql

script = """DROP TABLE IF EXISTS sysNext;
CREATE TABLE sysNext(
  next BIGINT not null
);
"""

def doesnt_work(cursor):
  cursor.execute(script)

def does_work(cursor):
  for x in script.split(';'):
    cmd = x.strip();
    if cmd != '':
      cursor.execute(cmd)

if __name__ == "__main__":
  connection = pymysql.connect(
    host='localhost',
    port=3306,
    user='root',
    charset='utf8',
    autocommit=True,
    use_unicode=True,
    no_delay=True
  ) 
  with connection as cur:
    cur.execute('DROP DATABASE IF EXISTS test;')
  with connection as cur:
    cur.execute('CREATE DATABASE test;')
  connection.select_db('test')
  with connection as cursor:
    ## call does_work(cursor) or doesnt_work(cursor) here
```
with does_work(cursor) I get warning 
```
>test_pymysql.bat
pymysql\cursors.py:134: Warning: Unknown table 'test.sysnext'
  result = self._query(query)
```

with doesnt_work(cursor) I get error:
```
>test_pymysql.bat
Traceback (most recent call last):
  File "test_pymysql.py", line 34, in <module>
    doesnt_work(cursor)
  File "test_pymysql.py", line 10, in doesnt_work
    cursor.execute(script)
  File "pymysql\cursors.py", line 134, in execute
    result = self._query(query)
  File "pymysql\cursors.py", line 283, in _query
    self._do_get_result()
  File "pymysql\cursors.py", line 298, in _do_get_result
    self._show_warnings(conn)
  File "pymysql\cursors.py", line 302, in _show_warnings
    for w in ws:
TypeError: 'NoneType' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test_pymysql.py", line 34, in <module>
    doesnt_work(cursor)
  File "pymysql\connections.py", line 754, in __exit__
    self.rollback()
  File "pymysql\connections.py", line 711, in rollback
    self._read_ok_packet()
  File "pymysql\connections.py", line 687, in _read_ok_packet
    raise err.OperationalError(2014, "Command Out of Sync")
pymysql.err.OperationalError: (2014, 'Command Out of Sync')
```

So maybe it's not multi-query that is the problem after all.

**Comment by methane** [comment]
Don't skip `cursor.close()`.  It is required for multi query.

PyMySQL introduced warning check in previous version.  It may cause regression when warning happen in middle of multi query. I'll see it when I have time.

**Comment by id-ilych** [comment]
Ok. Thanks.

About `cursor.close()`... How would I automate that using 'with' statement? 
`Connection.__enter__` produces cursor, and `__exit__` only do rollback or commit leaving cursor untouched. `Cursor` has `__enter__/__exit__` on it's own that closes it. So should i write something like that:
```
with connection as x, x as cursor:
  do_work(cursor)
```
to handle that automatically?

I did some more research. 

1. Without drop/create database commands script works w/o any warnings no matter of `cursor.close()` being called or not.
2. When using `with connection as x, x as cursor` second exception `OperationalError` is gone leaving me with just `TypeError: 'NoneType' object is not iterable`.

**Comment by methane** [comment]
> So should i write something like that:
>
> with connection as x, x as cursor:
>   do_work(cursor)
>
> to handle that automatically?

Yes, if you want to handle transaction and closing cursor in one line.
But I don't recommend to do it with one line. I think `Connection.__enter__` was bad idea.
It should be like following:

```python
with conn.begin():
    # ...
    with conn.cursor() as cur:
        cur.execute(...)
```

> Without drop/create database commands script works w/o any warnings no matter of cursor.close() being called or not.

Yes. `cur.nextset()` will be called next time. But you shouldn't rely on this behavior.
It makes difficult to investigate error when something goes wrong.

> When using with connection as x, x as cursor second exception OperationalError is gone leaving me with just TypeError: 'NoneType' object is not iterable.

Warning checking may have some bug.
I don't have time to research how can I check warning with multi statement for now.

**Comment by id-ilych** [comment]
OK :(

Off topic: `conn.begin()` doesn't seem to be a proper context manager right now so I guess you think about adding `return self` to `Connection.begin` and to replace `Connection.__enter__` with just `pass`. Is that correct?

**Comment by methane** [comment]
Yes. I said about what the API should be.

**Comment by methane** [comment]
fixed via 818edd83549b1433163f09c3b93e6024fcca4c03
